pkg/build/api/types.go

```diff
@@ -55,6 +55,9 @@ type BuildParameters struct {
 
 	// Output describes the Docker image the Strategy should produce.
 	Output BuildOutput `json:"output,omitempty"`
+
+	// Compute resource requirements to execute the build
+	Resources kapi.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
> On Apr 21, 2015, at 11:02 PM, Ben Parees <notifications@github.com> wrote:
> 
> In pkg/build/api/types.go:
> 
> > @@ -55,6 +55,9 @@ type BuildParameters struct {
> >  
> >  	// Output describes the Docker image the Strategy should produce.
> >  	Output BuildOutput `json:"output,omitempty"`
> > +
> > +	// Compute resource requirements to execute the build
> > +	Resources kapi.ResourceRequirements `json:"resources,omitempty"`
> (it's also a waste to allocate an empty object when a nil ptr would do the job)
> 
Adding if x != nil && len(x.limits) > 0 all over the code is also a waste.  It's also less succinct.
> —
> Reply to this email directly or view it on GitHub.
>